core/nwb.event.yaml

```
@@ -0,0 +1,53 @@
+groups:
+- neurodata_type_def: Events
+  neurodata_type_inc: NWBDataInterface
+  doc: A list of timestamps, stored in seconds, of an event
+  attributes:
+  - name: description
+    dtype: text
+    doc: Description of the event.
+  datasets:
+  - name: timestamps
+    dtype: float64
```

```suggestion
    dtype: float
```
I prefer `float`. This will allow the user to use `float32` or `float64`.

---

In the spec, I'm pretty sure `float` corresponds to `float32`. See https://schema-language.readthedocs.io/en/latest/specification_language_description.html#dtype and the corresponding code in https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/spec/spec.py#L22-L42 .

The timestamps of `TimeSeries` has dtype `float64` and I think timestamps should have higher precision in general.

---

The spec defines the *minimum allowable precision* so `float32` would allow you to use `float64`

---

Ah, I see that. Sorry, I missed that. What do you think about replacing `float64` with `float` for the other instances below?
TimeSeries starting_time
TimeSeries timestamps
SpikeEventSeries timestamps
FeatureExtraction times
EventDetection times

There are also 16 places where int32 and two places where uint64 is specified, most if not all of which do not require such high precision.

---

I'm in favor of flexible precision throughout. I can't see a downside